include/options.hxx

```diff
@@ -164,6 +164,21 @@ public:
   Options(Options *parent_instance, std::string full_name)
       : parent_instance(parent_instance), full_name(std::move(full_name)){};
 
+  /// Initialise with a value
+  /// These enable Options to be constructed using initializer lists
+  /// @{
+  template <typename T>
+  Options(T value) {
+    assign<T>(value);
+  }
+  /// @}
+  
+  /// Construct with a nested initializer list
+  /// This allows Options trees to be constructed, using a mix of types.
+  ///
+  /// Example:  { {"key1", 42}, {"key2", field} }
+  Options(std::initializer_list<std::pair<const std::string, const Options>> values);
```

Comment:
Is `const` needed here? I think I can see why one might be needed on the `std::string`, as the map `value_type` is `pair<const Key, T>`

Note: I don't think it hurts anything to keep it!